File: tests/src/main/java/com/github/klee0kai/test/mowgli/animal/Snake.java
package com.github.klee0kai.test.mowgli.animal;

import com.github.klee0kai.test.mowgli.body.Blood;
import com.github.klee0kai.test.mowgli.community.History;
import com.github.klee0kai.test.mowgli.galaxy.Earth;
import com.github.klee0kai.test.mowgli.identity.Conscience;
import com.github.klee0kai.test.mowgli.identity.Knowledge;

import javax.inject.Inject;

public class Snake implements IAnimal {

    @Inject
    public Blood blood;
    @Inject
    public Earth earth;
    @Inject
    public History history;
    @Inject
    public Conscience conscience;
    @Inject
    public Knowledge knowledge;

}

File: tests/src/commonMain/kotlin/com/github/klee0kai/test/mowgli/animal/Snake.kt
package com.github.klee0kai.test.mowgli.animal

import com.github.klee0kai.stone.Inject
import com.github.klee0kai.test.mowgli.body.Blood
import com.github.klee0kai.test.mowgli.community.History
import com.github.klee0kai.test.mowgli.galaxy.Earth
import com.github.klee0kai.test.mowgli.identity.Conscience
import com.github.klee0kai.test.mowgli.identity.Knowledge
import kotlin.jvm.JvmField

class Snake : IAnimal {
    @JvmField
    @Inject
    var blood: Blood? = null

    @JvmField
    @Inject
    var earth: Earth? = null

    @JvmField
    @Inject
    var history: History? = null

    @JvmField
    @Inject
    var conscience: Conscience? = null

    @JvmField
    @Inject
    var knowledge: Knowledge? = null
}
